### PR TITLE
Implement casting

### PR DESCRIPTION
**Pull request by luizParreira**
This PR implements the main functionality of the package.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,36 @@
 # EctoCastint
 
-**TODO: Add description**
+EctoCastint is a simple library that enables ecto casting from integer to string.
 
 ## Installation
 
 If [available in Hex](https://hex.pm/docs/publish), the package can be installed
-by adding `ecto_castint` to your list of dependencies in `mix.exs`:
+by adding `:eccto_castint` to your list of dependencies in `mix.exs`:
```
Comment (philss) [inline README.md]: type: `:ecto_castint`

File: mix.exs [hunk 2]
```diff
@@ -28,6 +44,6 @@ defmodule EctoCastint.Mixfile do
   #
   # Type "mix help deps" for more examples and options
   defp deps do
-    []
+    [{:phoenix_ecto, "~> 3.0"}]
```
Comment (philss) [inline mix.exs]: I think it should be only `ecto`. Is there a reason to be `phoenix_ecto`?

File: README.md [hunk 3]
```diff
@@ -1,19 +1,36 @@
 # EctoCastint
 
-**TODO: Add description**
+EctoCastint is a simple library that enables ecto casting from integer to string.
```
Comment (philss) [inline README.md]: I think the current name is not totally self-explained: it says `cast int`, but don't say what the cast we are doing.
 Do you agree to add an "string" in the name? Like `EctoCastIntToString`?

File: lib/ecto_castint.ex [hunk 4]
```diff
@@ -1,18 +1,52 @@
...
+  @spec cast(integer) :: {:ok, String.t} | :error
+  def cast(integer) when is_integer(integer), do: {:ok, Integer.to_string(integer)}
+
+  # We should still accept strings
+  def cast(string) when is_bitstring(string), do: {:ok, string}
+
+  # Everything else is a failure though
+  def cast(_), do: :error
+
+  def load(string) when is_bitstring(string), do: {:ok, string}
```
Comment (philss) [inline lib/ecto_castint.ex]: I miss the tests of this function.

File: lib/ecto_castint.ex [hunk 5]
```diff
@@ -1,18 +1,52 @@
...
+
+  # When dumping data to the database, we *expect* an astring
+  # but any value could be inserted into the struct, so we need
+  # guard against them.
+  def dump(string) when is_bitstring(string), do: {:ok, string}
```
Comment (philss) [inline lib/ecto_castint.ex]: and tests for this one too.

**Review by philss — CHANGES_REQUESTED** [review]
Added some points. We can discuss better the name.

File: README.md (hunk 3 above)
Comment (luizParreira) [reply README.md]: 98f17fa b58ae1f Agree 👍

File: lib/ecto_castint.ex (hunk 5 above)
Comment (luizParreira) [reply lib/ecto_castint.ex]: 98f17fa

File: lib/ecto_castint.ex (hunk 4 above)
Comment (luizParreira) [reply lib/ecto_castint.ex]: 98f17fa

File: mix.exs (hunk 2 above)
Comment (luizParreira) [reply mix.exs]: 98f17fa

**Review by philss — APPROVED** [review]

**Comment by philss** [comment]
:shipit: 

I recommend that you use `squash and merge` instead of simple merge.

**Pull request merged**